ui/app/scripts/shared/services/parser.js

```diff
@@ -700,13 +705,44 @@ define(function() {
 	        	}
         		node.errors.push(error);
         	}
+			
+			function eatTaskDefinition(lineNum) {
+				var taskNode = {};
+				var taskName = maybeEatStreamName();
```

Comment:
Good catch. I will make that generic and adjust the error it throws. I notice the insert in the message isn't right either, we need to include a precise field from the token, not just the token. You can trigger seeing that it is wrong by doing "==abc" and it will complain about the name "="